### PR TITLE
Add unwrap to Value<F> and make assign() pub

### PR DESCRIPTION
**Pull request by chaosma**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ Cargo.lock
 .vscode
 **/*.html
 .DS_Store
+*.png
```
Comment (drouyang) [inline .gitignore]: remove from PR

File: halo2_gadgets/src/poseidon/pow5.rs [hunk 2]
```diff
@@ -161,6 +161,7 @@ impl<F: Field, const WIDTH: usize, const RATE: usize> Pow5Chip<F, WIDTH, RATE> {
         });
 
         meta.create_gate("pad-and-add", |meta| {
+            // chao: last element is capacity element here
```
Comment (drouyang) [inline halo2_gadgets/src/poseidon/pow5.rs]: remove from PR

File: halo2_proofs/benches/arithmetic.rs [hunk 3]
```diff
@@ -1,36 +1,48 @@
 #[macro_use]
```
Comment (drouyang) [inline halo2_proofs/benches/arithmetic.rs]: Could you create a separate PR for this benchmark change?

File: halo2_proofs/src/circuit/value.rs [hunk 4]
```diff
@@ -24,6 +24,10 @@ impl<V> Default for Value<V> {
 }
```
Comment (drouyang) [inline halo2_proofs/src/circuit/value.rs:24]: This change worth a dedicated PR

File: halo2_proofs/src/circuit/value.rs [hunk 5]
```diff
@@ -45,7 +49,7 @@ impl<V> Value<V> {
     /// Obtains the inner value for assigning into the circuit.
     ///
     /// Returns `Error::Synthesis` if this is [`Value::unknown()`].
-    pub(crate) fn assign(self) -> Result<V, Error> {
```
Comment (chaosma) [inline halo2_proofs/src/circuit/value.rs:48]: this function is used in sirius:

```
// src/plonk.rs

fn assign_fixed<V, VR, A, AR>(
        &mut self,
        _: A,
        column: Column<Fixed>,
        row: usize,
        to: V,
    ) -> Result<(), Error>
    where
        V: FnOnce() -> Value<VR>,
        VR: Into<Assigned<F>>,
        A: FnOnce() -> AR,
        AR: Into<String>,
    {
        *self
            .fixed
            .get_mut(column.index())
            .and_then(|v| v.get_mut(row))
            .ok_or(Error::BoundsFailure)? = to().into_field().assign()?;
        Ok(())
    }

```

File: halo2_proofs/src/dev.rs [hunk 6]
```diff
@@ -818,7 +818,21 @@ impl<F: FromUniformBytes<64> + Ord> MockProver<F> {
                                 &Value::Real(F::ZERO),
```
Comment (drouyang) [inline halo2_proofs/src/dev.rs]: Are changes here for debugging?

**Review by drouyang — CHANGES_REQUESTED** [review]

File: halo2_proofs/src/circuit/value.rs [hunk 7]
```diff
@@ -24,6 +24,10 @@ impl<V> Default for Value<V> {
 }
 
 impl<V> Value<V> {
+    /// unwrap the inner field 
+    pub const fn unwrap(&self) -> &Option<V> {
```
Comment (chaosma) [inline halo2_proofs/src/circuit/value.rs:28]: this is used for better handle Value<V>

**Review by chaosma — COMMENTED** [review]
There are only two necessary changes which explained in the comments, other part of codes are either for better debug message or benchmark purpose.

[update] remove unnecessary changement.

File: halo2_proofs/src/dev.rs (hunk 6 above)
Comment (chaosma) [reply halo2_proofs/src/dev.rs]: removed

File: halo2_proofs/benches/arithmetic.rs (hunk 3 above)
Comment (chaosma) [reply halo2_proofs/benches/arithmetic.rs]: removed

File: halo2_gadgets/src/poseidon/pow5.rs (hunk 2 above)
Comment (chaosma) [reply halo2_gadgets/src/poseidon/pow5.rs]: removed

File: .gitignore (hunk 1 above)
Comment (chaosma) [reply .gitignore]: removed

**Review by drouyang — APPROVED** [review]

**Pull request merged**